src/EnergyPlus/ZoneTempPredictorCorrector.cc

```diff
@@ -3708,267 +3443,146 @@ void PredictSystemLoads(EnergyPlusData &state,
                 } else {
-                    ZoneT1(ZoneNum) = state.dataHeatBalFanSys->ZoneTMX(ZoneNum);
-                    state.dataHeatBalFanSys->ZoneW1(ZoneNum) = state.dataHeatBalFanSys->ZoneWMX(ZoneNum);
-                    if (state.dataRoomAirMod->AirModel(ZoneNum).AirModelType == DataRoomAirModel::RoomAirModel::AirflowNetwork) {
-                        for (LoopNode = 1; LoopNode <= RoomAirflowNetworkZoneInfo(ZoneNum).NumOfAirNodes; ++LoopNode) {
-                            RoomAirflowNetworkZoneInfo(ZoneNum).Node(LoopNode).AirTempT1 =
-                                RoomAirflowNetworkZoneInfo(ZoneNum).Node(LoopNode).AirTempTMX;
-                            RoomAirflowNetworkZoneInfo(ZoneNum).Node(LoopNode).HumRatW1 =
-                                RoomAirflowNetworkZoneInfo(ZoneNum).Node(LoopNode).HumRatWMX;
-                        }
-                    }
+                    thisTempControlledZone.CoolModeLast = false;
                 }
-                state.dataHVACGlobal->ShortenTimeStepSysRoomAir = true;
-            } else {
-                ZoneT1(ZoneNum) = state.dataHeatBalFanSys->ZT(ZoneNum);
-                state.dataHeatBalFanSys->ZoneW1(ZoneNum) = state.dataHeatBalFanSys->ZoneAirHumRat(ZoneNum);
-                if (state.dataRoomAirMod->AirModel(ZoneNum).AirModelType == DataRoomAirModel::RoomAirModel::AirflowNetwork) {
-                    for (LoopNode = 1; LoopNode <= RoomAirflowNetworkZoneInfo(ZoneNum).NumOfAirNodes; ++LoopNode) {
-                        RoomAirflowNetworkZoneInfo(ZoneNum).Node(LoopNode).AirTempT1 = RoomAirflowNetworkZoneInfo(ZoneNum).Node(LoopNode).AirTemp;
-                        RoomAirflowNetworkZoneInfo(ZoneNum).Node(LoopNode).HumRatW1 = RoomAirflowNetworkZoneInfo(ZoneNum).Node(LoopNode).HumRat;
-                    }
+                if (thisTempControlledZone.HeatOffFlag && state.dataZoneEnergyDemand->ZoneSysEnergyDemand(ZoneNum).TotalOutputRequired <= 0.0) {
+                    thisTempControlledZone.HeatModeLast = true;
+                } else {
+                    thisTempControlledZone.HeatModeLast = false;
                 }
             }
-            TempDepZnLd(ZoneNum) = TempDepCoef;
-            TempIndZnLd(ZoneNum) = TempIndCoef;
         }
+    }
+}
+void ZoneSpaceHeatBalanceData::predictSystemLoad(
+    EnergyPlusData &state,
+    bool const shortenTimeStepSys,
+    bool const useZoneTimeStepHistory, // if true then use zone timestep history, if false use system time step
+    Real64 const priorTimeStep,        // the old value for timestep length is passed for possible use in interpolating
+    int zoneNum,
+    int spaceNum)
+{
+    assert(zoneNum > 0);
+    this->updateTemperatures(state, shortenTimeStepSys, useZoneTimeStepHistory, priorTimeStep, zoneNum, spaceNum);
+
+    Real64 volume = 0.0;
+    if (spaceNum > 0) {
+        volume = state.dataHeatBal->space(spaceNum).Volume;
+    } else {
+        volume = state.dataHeatBal->Zone(zoneNum).Volume;
+    }
```

Comment:
For data that's outside the `*HeatBalance` struct, test on `spaceNum` to grab info from zone or space data.